### PR TITLE
[GOBBLIN-743] Initialize Gobblin application master services with dyn…

### PR DESCRIPTION
**Pull request by htran1**
…amic config

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-743

### Description
- [X] Here are some details about my PR, including screenshots (if applicable):
The Gobblin application manager needs to initialize services with the config generated by the dynamic config generator. One use case that requires this is the passing of SSL configuration to kafka consumers and producers.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
GobblinYarnAppLauncherTest

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java [hunk 1]
```diff
@@ -17,8 +17,12 @@
 
 package org.apache.gobblin.yarn;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.io.Closer;
+import com.google.common.util.concurrent.Service;
+import com.sun.source.tree.AssertTree;
```
Comment (shirshanka) [inline gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java]: unused import?

File: gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java [hunk 2]
```diff
@@ -156,32 +172,27 @@ public void setUp() throws Exception {
...
 
   @Test
-  public void testCreateHelixCluster() throws Exception {
+  public void testCreateHelixCluster()
+      throws Exception {
     // This is tested here instead of in HelixUtilsTest to avoid setting up yet another testing ZooKeeper server.
-    HelixUtils.createGobblinHelixCluster(
-        this.config.getString(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY),
-        this.config.getString(GobblinClusterConfigurationKeys.HELIX_CLUSTER_NAME_KEY));
+    HelixUtils
```
Comment (shirshanka) [inline gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java]: formatting seems weird here

Comment (htran1) [reply gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java]: Yeah, missed that the IDE auto formatted this file. Reverted the autoformat changes.

File: gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java (hunk 1 above)
Comment (htran1) [reply gobblin-yarn/src/test/java/org/apache/gobblin/yarn/GobblinYarnAppLauncherTest.java]: Yes, removed this.

**Pull request closed without merging**